guava/src/com/google/common/cache/StatsCounter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2011 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.common.cache;
+
+import com.google.common.annotations.GwtCompatible;
+
+/**
+ * Accumulates statistics during the operation of a {@link Cache} for presentation by {@link
+ * Cache#stats}. This is solely intended for consumption by {@code Cache} implementors.
+ *
+ * @since 10.0
+ */
+@GwtCompatible
+public interface StatsCounter {
```

Comment:
It would probably need to extend the original, with that marked as deprecated. Then it could go through the graceful removal process.